src/library/dao/trackdao.cpp

```diff
@@ -1227,7 +1227,20 @@ TrackPointer TrackDAO::getTrackFromDB(TrackId trackId) const {
         // Just to be safe, but this should never happen!!
         return pTrack;
     }
-    DEBUG_ASSERT(pTrack->getId() == trackId);
+    if (pTrack->getId() != trackId) {
+        // This happens if two different tracks are referencing
```

Comment:
// The cacheResolver() failed to resolve the track by it's trackId, but has found a different track listed in the database, referencing ...

It took me a bit to remind me how everything is connected. So I think I think this comment helps here.

Reply:
The issue is that the using code keeps has the old track ID already on stack and keeps using it. 
So we either need to inform the using code about the changed track ID or fail here.

Reply:
I am not able to solve all downstream issues and also haven't spotted any serious issues at first sight.